File: 310971/Stankevich/cassandra330/publisher/src/main/java/com/task/rest/dto/NoticeRequestTo.java
package com.task.rest.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class NoticeRequestTo {

    private Long id;

    @Size(min = 2, max = 64)
    private String country;

    @NotNull
    private Long tweetId;

    @NotBlank
    @Size(min = 2, max = 2048)
    private String content;
}

File: 351003/brezgunov/distcomp/src/main/kotlin/com/distcomp/dto/notice/NoticeRequestTo.kt
package com.distcomp.dto.notice

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size
import java.util.UUID

data class NoticeRequestTo (
    val id: UUID? = null,

    val country: String? = null,

    @field:NotNull
    var newsId: Long,

    @field:NotBlank
    @field:Size(min = 4, max = 2048)
    val content: String
)